apps/vue-showcase/cypress/e2e/vue-route-guard.cy.ts

```diff
@@ -0,0 +1,73 @@
+const GUARD_TOKEN_NAME = 'VUEROUTEGUARD-TOKEN';
+
+describe(`@this-dot/vue-route-guard`, () => {
+  describe(`No authentication set`, () => {
+    before(() => {
+      cy.window().then((win) => {
+        win.sessionStorage.removeItem(GUARD_TOKEN_NAME);
+      });
+      cy.clearLocalStorage(GUARD_TOKEN_NAME);
+      cy.clearCookie(GUARD_TOKEN_NAME);
+    });
+
+    it(`redirects to login when user visits a guarded page`, () => {
+      cy.visit('/route-guard');
+
+      cy.url().should('contain', 'route-guard/login');
+    });
+
+    it(`does not redirect when user visits unguarded page`, () => {
+      cy.visit('/route-guard/no-permission');
+
+      cy.url().should('contain', '/route-guard/no-permission');
+    });
+
+    it(`does not redirect to home view when home navigation is clicked`, () => {
+      cy.visit('/route-guard');
+
+      cy.get(`[data-test-id="home navigation button"]`).should('be.visible').click();
+      cy.url().should('not.eq', Cypress.config().baseUrl + '/route-guard');
+    });
+
+    it(`about navigation is not displayed`, () => {
+      cy.visit('/route-guard');
+
+      cy.get(`[data-test-id="about navigation button"]`).should('not.exist');
+    });
+  });
+
+  describe(`Authentication set`, () => {
+    before(() => {
+      cy.window().then((win) => {
+        win.sessionStorage.setItem(GUARD_TOKEN_NAME, 'sample_token');
+      });
+    });
+
+    it(`does not redirect when user visits guarded page`, () => {
+      cy.visit('/route-guard');
+      cy.url().should('eq', Cypress.config().baseUrl + '/route-guard');
+    });
+
+    it(`authentication data is set in session storage`, () => {
```

Comment:
This test is kind-of redundant, because you set it in the before hook. This test would test the implementation logic of the login button, but now it has no value. Please remove it

Reply:
Thank for catching that.